bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschSHCHandler.java

```diff
@@ -120,6 +135,11 @@ public void processUpdate(String serviceName, JsonElement stateData) {
         }
     }
 
+    /**
+     * Returns the bridge handler for this thing handler.
+     * 
+     * @return Bridge handler for this thing handler. Null if no or an invalid bridge was set in the configuration.
+     */
     protected @Nullable BoschSHCBridgeHandler getBridgeHandler() {
         Bridge bridge = this.getBridge();
         if (bridge == null) {
```

Comment:
Suggestion for refactoring:
We can throw here an exception like

    throw new Error(String.format("Could not initialize {}, no valid bridge set", this.getThing()));

and avoid several null checks in derived classes like the one in ``ThermostatHandler::initialize()``

Reply:
That's a good idea, I will try that. We will still have to catch the exception during initialization as the bridge is set by the user (e.g. in Paper UI) and we have to show a user-friendly error in that case instead of having only an error in the openHAB log.

But at least this might make additional null checks after the initialization of a handler obsolete.